src/query/api/v1/handler/prometheus/native/read.go

```diff
@@ -75,10 +82,15 @@ type blockWithMeta struct {
 func NewPromReadHandler(
 	engine *executor.Engine,
 	tagOpts models.TagOptions,
-) http.Handler {
+	limitsCfg *config.LimitsConfiguration,
+) *PromReadHandler {
+	if limitsCfg == nil {
```

Comment:
do this in the server.go method instead of the handler ? I'd assume other handlers might need the limitsconfiguration too

Reply:
Good call, will move.

Reply:
Hmm looking at `server.go`, there's not a super obvious place in `Run` to put it. Should I start an `applyDefaults` function or something to that effect? Would be good to centralize that.

Reply:
I put just above handler construction for now--probably not ideal, so lmk if you have any suggestions on that front.